src/wp-includes/fonts/class-wp-font-family.php

```diff
@@ -0,0 +1,615 @@
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $font_face The font face array containing the 'src' attribute
+	 *                         with the file path(s) to be deleted.
+	 * @return bool True if delete was successful, otherwise false.
+	 */
+	private static function delete_font_face_assets( $font_face ) {
+		$sources = (array) $font_face['src'];
+		foreach ( $sources as $src ) {
+			$was_asset_removed = self::delete_asset( $src );
+			if ( ! $was_asset_removed ) {
+				// Bail if any of the assets could not be removed.
+				return false;
+			}
+		}
+		return true;
+	}
+
+
+	/**
+	 * Gets the overrides for the 'wp_handle_upload' function.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $filename The filename to be used for the uploaded file.
+	 * @return array The overrides for the 'wp_handle_upload' function.
+	 */
+	private function get_upload_overrides( $filename ) {
+		return array(
+			// Arbitrary string to avoid the is_uploaded_file() check applied
+			// when using 'wp_handle_upload'.
+			'action'                   => 'wp_handle_font_upload',
+			// Not testing a form submission.
+			'test_form'                => false,
+			// Seems mime type for files that are not images cannot be tested.
+			// See wp_check_filetype_and_ext().
+			'test_type'                => true,
+			'mimes'                    => WP_Font_Library::get_expected_font_mime_types_per_php_version(),
+			'unique_filename_callback' => static function () use ( $filename ) {
+				// Keep the original filename.
+				return $filename;
+			},
+		);
+	}
+
+	/**
+	 * Downloads a font asset from a specified source URL and saves it to
+	 * the font directory.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $url      The source URL of the font asset to be downloaded.
+	 * @param string $filename The filename to save the downloaded font asset as.
+	 * @return string|bool The relative path to the downloaded font asset.
+	 *                     False if the download failed.
+	 */
+	private function download_asset( $url, $filename ) {
+		// Checks if the file to be downloaded has a font mime type.
+		if ( ! WP_Font_Family_Utils::has_font_mime_type( $filename ) ) {
+			return false;
+		}
+
+		// Include file with download_url() if function doesn't exist.
+		if ( ! function_exists( 'download_url' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+		}
+
+		// Downloads the font asset or returns false.
+		$temp_file = download_url( $url );
+		if ( is_wp_error( $temp_file ) ) {
+			return false;
+		}
+
+		$overrides = $this->get_upload_overrides( $filename );
+
+		$file = array(
+			'tmp_name' => $temp_file,
+			'name'     => $filename,
+		);
+
+		$handled_file = wp_handle_upload( $file, $overrides );
+
+		// Cleans the temp file.
+		@unlink( $temp_file );
+
+		if ( ! isset( $handled_file['url'] ) ) {
+			return false;
+		}
+
+		// Returns the relative path to the downloaded font asset to be used as
+		// font face src.
```

Comment:
```suggestion
		/*
		 * Returns the relative path to the downloaded font asset to be used as
		 *  font face 'src'.
		 */
```
- See above regarding the multi-line comment format.
- Also added quotes around the `src` key.

Reply:
Committed in https://github.com/WordPress/wordpress-develop/pull/5285/commits/e948c1b764370bba2fa982b190e3c9dbb179cfab ✅